src/ros_filter.cpp

```diff
@@ -90,6 +90,39 @@ namespace RobotLocalization
     topicSubs_.clear();
   }
 
+  template<typename T>
+  void RosFilter<T>::reset()
+  {
+    // Get rid of any initial poses (pretend we've never had a measurement)
+    initialMeasurements_.clear();
+    previousMeasurements_.clear();
+    previousMeasurementCovariances_.clear();
+
+    // Clear the measurement queue.
+    // This prevents us from immediately undoing our reset.
+    while (!measurementQueue_.empty() && ros::ok())
+    {
+      measurementQueue_.pop();
+    }
+
+    ros::getGlobalCallbackQueue()->clear();
```

Comment:
http://docs.ros.org/diamondback/api/roscpp/html/classros_1_1CallbackQueue.html#ab960789600b6a6d775249347c234019a

I've also experienced it. If you use this method from within a callback, there's no guarantee the rest of the callback will  execute. I need to fix the other call.